src/huggingface_hub/file_download.py

```diff
@@ -514,6 +526,11 @@ def cached_download(
 
     </Tip>
     """
+    warnings.warn(
+        "`cached_download` is the legacy way to download files from the HF hub, please"
+        " consider upgrading to `hf_hub_download`",
+        FutureWarning,
+    )
```

Comment:
@julien-c, `cached_download` implements features that are not available with `hf_hub_download` (namely, caching from a URL). Is this a feature that you think should be abandonned? I think there is room for both methods to exist, but this warning says otherwise.

Reply:
> Is this a feature that you think should be abandonned?

I think it's a feature that does not really make sense in the context of the HF hub, and a HF Hub client library.

My concern is that if we don't add this warning, users will not upgrade to the new way.

As a middleground, maybe we could add a `legacy_cache_layout` flag that just disables the warning? and mention in the warning "If you want to cache from an arbitrary URL instead, pass legacy_cache_layout=True"

WDYT?

Reply:
Yes, sounds good to me. I'll proceed this way.